scriptworker/test/test_worker.py

```diff
@@ -254,3 +260,296 @@ def _mocker_run_tasks_helper(mocker, exc, func_to_raise):
     mocker.patch.object(worker, "complete_task", new=noop_async)
     status = await worker.run_tasks(context)
     assert status == context.config['task_max_timeout_status']
+
+
+_MOCK_CLAIM_WORK_RETURN = {
+    'tasks': [{
+        # don't need to worry about the contents of each task for these tests
+    }]
+}
+
+_MOCK_CLAIM_WORK_NONE_RETURN = {
+    'tasks': []
+}
+
+
+class MockChainOfTrust:
+    def __init__(self, context, cot_job_type):
+        pass
+
+
+class MockTaskProcess:
+    def __init__(self):
+        self.stopped_due_to_worker_shutdown = False
+        self.worker_stop_future = asyncio.Future()
+
+    async def worker_shutdown_stop(self):
+        self.stopped_due_to_worker_shutdown = True
+        self.worker_stop_future.set_result(None)
+
+    async def _wait(self):
+        await self.worker_stop_future
+
+
+def create_finished_promise(result=None):
+    future = asyncio.Future()
+    future.set_result(result)
+    return future
+
+
+def create_slow_async():
+    future = asyncio.Future()
+
+    async def slow_function(*args, **kwargs):
+        future.set_result(None)
+        await asyncio.Future()
+
+    return future, slow_function
+
+
+def create_async(result=None):
+    async def fn(*args, **kwargs):
+        return result
+    return fn
+
+
+def create_sync(result=None):
+    def fn(*args, **kwargs):
+        return result
+    return fn
+
+
+@pytest.mark.asyncio
+async def test_run_tasks_no_cancel(context, mocker):
+    mocker.patch('scriptworker.worker.claim_work', create_async(_MOCK_CLAIM_WORK_RETURN))
+    mocker.patch.object(asyncio, 'sleep', create_async())
+    mocker.patch('scriptworker.worker.prepare_to_run_task', create_async())
+    mocker.patch('scriptworker.worker.reclaim_task', create_async())
```

Comment:
I've got `noop_async` and `noop_sync` sprinkled throughout the test code, possible these are combinable.

Also, agreed that `test_worker.py` has a *ton* of mocks. But even if we follow clean architecture best practices, somewhere we're dealing with the i/o and complexity. In scriptworker, that's `scriptworker.worker`.